runtime/core/src/entities/factory.js

```diff
@@ -11,6 +13,10 @@ class Factory {
     this.#schema = schema
   }
 
+  fit (values) {
+    this.#schema.validate({ id: newid(), ...values }, 'Entity')
+  }
```

Comment:
NIT: not related to current PR but MongoDB has built-in support for JSON Schema that may guarantee schema fit at storage level. Going forward we should probably prefer db-side validation and use toa-side only when DB can't do so.

Reply:
1. I can't see any benefits in sending data over a network to validate it.
2. There are (and will be) scenarios, where data is guaranteed to be already valid (e.g. data synchronization), so validation will do nothing except wasting resources.
3. This is the core, which is abstracted from the specific data storage, therefore relying on storage implementation will increase an amount of requirements to storage implementers.